<!DOCTYPE html> <html> <head> <meta name="google-site-verification" content="google-site-verification=google5d0dde76b59b2bd0.html"/> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>Evolutionary Scale Modeling using Protein Language Models | Amitesh Badkul</title> <meta name="author" content="Amitesh Badkul"/> <meta name="description" content="ESM2 model explained"/> <meta name="keywords" content="deep learning, bioinformatics, machine learning, image processing"/> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"/> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light"/> <link rel="shortcut icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>ðŸ’­</text></svg>"> <link rel="stylesheet" href="/assets/css/main.css"> <link rel="canonical" href="https://amiteshbadkul.github.io/blog/2023/esm2-explained/"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark"/> <script src="/assets/js/theme.js"></script> <script src="/assets/js/dark_mode.js"></script> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src="/assets/js/distillpub/template.v2.js"></script> <script src="/assets/js/distillpub/transforms.v2.js"></script> <script src="/assets/js/distillpub/overrides.js"></script> </head> <d-front-matter> <script async type="text/json">{
      "title": "Evolutionary Scale Modeling using Protein Language Models",
      "description": "ESM2 model explained",
      "published": "July 29, 2023",
      "authors": [
        {
          "author": "Amitesh Badkul",
          "authorURL": "https://amiteshbadkul.github.io",
          "affiliations": [
            {
              "name": "City University of New York",
              "url": ""
            }
          ]
        }
        
      ],
      "katex": {
        "delimiters": [
          {
            "left": "$",
            "right": "$",
            "display": false
          },
          {
            "left": "$$",
            "right": "$$",
            "display": true
          }
        ]
      }
    }</script> </d-front-matter> <body class="fixed-top-nav"> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="https://amiteshbadkul.github.io/"><span class="font-weight-bold">AmiteshÂ </span>Badkul</a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about</a> </li> <li class="nav-item active"> <a class="nav-link" href="/blog/">blog<span class="sr-only">(current)</span></a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications</a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects</a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories</a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv</a> </li> <li class="toggle-container"> <button id="search-toggle" title="Search"> <i class="fas fa-search"></i> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fas fa-moon"></i> <i class="fas fa-sun"></i> </button> </li> </ul> </div> </div> </nav> </header> <div class="post distill"> <d-title> <h1>Evolutionary Scale Modeling using Protein Language Models</h1> <p>ESM2 model explained</p> </d-title> <d-byline></d-byline> <d-article> <d-contents> <nav class="l-text figcaption"> <h3>Contents</h3> <div><a href="#language-models-in-biological-domain">Language Models in Biological Domain</a></div> <div><a href="#protein-language-models">Protein Language Models</a></div> <div><a href="#emergence-of-structural-information-in-esm-2">Emergence of Structural Information in ESM-2</a></div> <div><a href="#esmfold-architecture">ESMFold Architecture</a></div> <div><a href="#conclusions">Conclusions</a></div> </nav> </d-contents> <h2 id="language-models-in-biological-domain">Language Models in Biological Domain</h2> <p>Language models like BERT have shown impressive capabilities for natural language tasks, inspiring their application to model protein sequences. These protein language models (PLMs) treat amino acid sequences analogous to sentences, with patterns reflecting structural and functional constraints. For instance, local amino acid order directly specifies secondary structure, while distal coevolving residues often contact in 3D space. Local amino acid order patterns such as helices and strands dictate the resulting local protein secondary structure. Meanwhile, amino acids distant in sequence but close spatially often coevolve by compensatory mutations to maintain structural integrity.</p> <p>PLMs are commonly pretrained via objectives like masked language modeling (MLM), which trains the model to predict randomly masked tokens from context. Despite simple self-supervision, large PLMs exhibit emergent understanding of language semantics. This motivated hypothesis that pretraining PLMs on protein corpora may similarly elicit learning of sequence-structure mappings, given patterns in sequences encode structural properties. Because patterns in protein sequences inherently reflect structural constraints, pretraining large PLMs to model sequences may enable learning about the sequence-structure relationship. This could allow PLMs to infer 3D structural properties purely from sequence, despite no explicit supervision.</p> <p>Critically, insights from statistical mechanics and coevolutionary sequence analysis established that proteinsâ€™ 3D shapes constrain viable amino acid substitutions over evolution. Thus structural information becomes embedded in sequence statistics. Early bioinformatics methods leveraged this to predict aspects like secondary structure or residue contacts.</p> <p>Machine learning advanced contact and structure prediction. Initial methods employed convolutional or recurrent architectures. Attention mechanisms then enabled modeling long-range dependencies, improving contact prediction. Recently, large multitask networks integrating both sequence and structure data have achieved highly accurate structure prediction.</p> <p>In parallel, natural language processing advanced via scaling model size and data. Self-attention-based transformers pretrained on text via MLM exhibit strong few-shot learning abilities. ProtBert first adapted BERT to proteins, showing MLM pretraining improved generalization. Subsequent PLMs including ProtT5 and ESM-1b demonstrated increasing capability to capture protein properties given sufficient scale.</p> <p>Scaling up MLM pretraining on large, diverse protein corpora may thus enable PLMs to implicitly learn mappings between sequence statistics and 3D conformations. This contrasts classic template-based or homology modeling, which compares query sequences against evolutionary relatives with known structures. PLMs can potentially associate sequences across fold space by identifying commonalities in sequence patterns induced by shared structural constraints. Realizing this possibility motivates recent research on large protein LM scaling.</p> <p>Overall, the convergence of statistical mechanics, coevolutionary theory, machine learning, and natural language processing over decades has laid the foundation for protein language models to potentially unlock new knowledge and capabilities as models scale. ESM-2 represents the latest innovation in this trajectory.</p> <h2 id="protein-language-models">Protein Language Models</h2> <p>Protein language models provide a new paradigm for learning representations integrating protein sequencesâ€™ structural and functional signatures. The core philosophy behind PLMs recognizes that sequences inherently encode signatures of 3D structures and biochemical activities into the order and variation of amino acids. This principle underpins advances in template modeling, physics-based simulations, and machine learned structure prediction.</p> <p>Despite straightforward pretraining objectives like MLM, large LMs demonstrate impressive zero-shot emergence of capabilities requiring understanding language semantics. For instance, models trained simply to fill masked words can exhibit nuanced skills like translation, summarization, and reasoning. Analogously, PLMs pretrained to predict masked amino acids from surrounding context may induce generalizable learning of sequence-structure relationships by internalizing meaningful sequence patterns.</p> <p>By pretraining on large, evolutionarily diverse sequence corpora, PLMs can potentially learn structural knowledge exceeding what is extractable from single protein families. PLMs complement traditional comparative modeling, which analyzes sequence covariation within groups of homologous proteins related by evolution. Modeling sequences collectively across protein space facilitates learning about fundamental physico-chemical constraints between sequences and structures.</p> <p>Recent PLMs exhibit promising capabilities for sequence-based structure prediction. For instance, models with over 10 billion parameters combined with differentiable folding heads can generate atomistic structural models from sequence alone. Some PLMs have achieved major speedups compared to previous methods while maintaining accuracy. Expanding protein sequence databases via metagenomic sequencing presents exciting opportunities for enlarging the scope of structures PLMs can effectively model.</p> <p>Overall, the ability of large neural language models to exhibit emergent capabilities has inspired approaches scaling up pretraining on protein sequences. ESM-2 substantiates the potential for this strategy to yield models encoding rich structural knowledge within sequence representations. Ongoing growth in model scale and available sequences promises continued development in protein language modeling.</p> <h2 id="emergence-of-structural-information-in-esm-2">Emergence of Structural Information in ESM-2</h2> <p>The ESM-2 model demonstrates substantial improvements in pretraining scale enables progressive emergence of protein structural properties in language model representations. Both conceptual insights into sequence-structure mappings and empirical experience with language models contributed to ESM-2â€™s development.</p> <table> <thead> <tr> <th style="text-align: center"><img src="https://imgur.com/0YIhOEp.png" width="75%" height="75%"></th> </tr> </thead> <tbody> <tr> <td style="text-align: center"><em>ESM-2 Model</em></td> </tr> </tbody> </table> <p>Theoretical principles established proteinsâ€™ 3D conformations constrain viable amino acid substitutions over evolution, embedding structural signatures within sequence statistics. This motivated pioneering bioinformatics approaches to leverage statistics for structure and function insight. Meanwhile, large neural language models, despite simple objectives like masked LM pretraining, exhibited surprising emergent capabilities requiring understanding language semantics.</p> <p>ESM-2 synthesizes these perspectives. It adopts a 24-layer transformer encoder architecture, with modifications including rotary position embeddings to better model long protein sequences. ESM-2 is pretrained on 65 million unique UniRef50 sequences using masked LM, enabling the model to learn sequence patterns and their structural implications.</p> <p>Analyses reveal consistent improvements in structural knowledge as ESM-2 scaling progresses from 8 million to 15 billion parameters. Both low-resolution and atomic-level structure quality metrics substantially increase with model size. Perplexity measurements also improve, confirming scaling enhances sequence representations. Critically, proteins exhibiting perplexity gains also show structure prediction gains, evidencing intimate connections between sequence modeling and structure learning. As ESM-2 scaling progressed from 8 million to 15 billion parameters, long-range contact precision increased substantially from 0.16 to 0.54 (238% relative gain). This metric measuring how accurately residue contacts can be extracted from self-attention maps improved steadily with model size. Similarly, CASP14 TM-score rose from 0.37 to 0.55 (49% relative gain), indicating atomic-level structure quality improved with scale. Perplexity declined from 10.45 to 6.37, confirming language modeling of sequences enhanced across scaling. Critically, proteins with large perplexity gains also showed large contact prediction gains (NDCG = 0.87), evidencing tight coupling between sequence modeling and structure prediction quality.</p> <p>Together, these results substantiate the hypothesis that massive self-supervision on diverse protein sequences can elicit emergence of structural knowledge within sequence representations. This supports the promise of large language model pretraining for deducing 3D conformations from sequence statistics by integrating co-evolutionary signals across protein space.</p> <p>The innovations of ESM-2 provide a foundation to further scale model capacity, sequence data, and compute power. Ongoing growth along these dimensions will offer insight into the limits of structural knowledge extractable purely from unsupervised sequence modeling.</p> <h2 id="esmfold-architecture">ESMFold Architecture</h2> <p>Building on insights from ESM-2, the ESMFold model enables fast, end-to-end protein structure prediction directly from sequence. ESMFold integrates the pretrained ESM-2 language model with a lightweight prediction head module. This simple yet powerful approach eliminates the need for homology search or complex multitask architectures.</p> <p>ESM-2 provides a strong foundation, having accumulated rich structural knowledge from its masked LM pretraining. Its representations are fed into ESMFoldâ€™s prediction module, which applies alternating layers of self- and pairwise attention to refine coordinate predictions. A triangular update scheme helps capture long-range dependencies, while maintaining efficiency.</p> <p>This overall architecture optimizes prediction speed without sacrificing accuracy. By exploiting the compressed evolutionary signal within ESM-2, costly alignment procedures are avoided. The prediction module also eliminates much complexity from state-of-the-art pipelines. Together, these savings allow up to 60x faster inference than existing methods while achieving competitive accuracy.</p> <p>ESM-2â€™s training is critical to ESMFoldâ€™s performance. Masked LM over 65+ million diverse sequences enables the language model to learn generalizable associations between sequence patterns and structural constraints. This rich pretraining endows ESMFold with single sequence prediction abilities exceeding alignment-dependent alternatives when alignments are ablated.</p> <p>Furthermore, ESM-2 perplexity predicts ESMFold accuracy, since perplexity improvements imply enhanced sequence representations that provide ESMFold superior structural insight. Overall, ESMFoldâ€™s design elegantly combines the power of scale with architectural simplicity to advance protein structure prediction.</p> <table> <thead> <tr> <th style="text-align: center"><img src="https://imgur.com/UKcvZas.png" width="75%" height="75%"></th> </tr> </thead> <tbody> <tr> <td style="text-align: center"><em>ESM-2 Model Perplexity</em></td> </tr> </tbody> </table> <p>The innovations of ESMFold highlight the potential for pretrained language models to accelerate structure prediction. Its speed and accuracy could expand high-quality prediction to large databases like metagenomic resources. Future work on integrating coevolutionary signals and physics-based refinement may further improve accuracy.</p> <h2 id="conclusions">Conclusions</h2> <p>In conclusion, ESM-2 and ESMFold demonstrate transformative potential for large language model pretraining in protein structure prediction. By pretraining on massive and diverse protein sequence corpora, representations emerge that encapsulate surprisingly detailed structural knowledge purely from sequence signals. This knowledge enables fast yet accurate end-to-end structure prediction from sequence alone.</p> <p>ESM-2 establishes clear trends of improving structural understanding with greater model scale and data diversity. Ongoing expansion of sequence databases via metagenomics promises continued gains from larger pretraining resources. In tandem, architectural advances may further enhance sequence representations.</p> <p>Looking forward, large language models like ESM-2 offer an exciting path to democratize structure prediction by distilling vast structural knowledge from sequences alone. Realizing this vision could accelerate discovery of new protein structures, functions, and mechanisms across biology. More broadly, scaling self-supervision may prove a general paradigm for eliciting emergence of complex reasoning abilities in artificial intelligence systems.</p> <p>The insights gained from ESM-2 and ESMFold underscore both the progress enabled by scaled language model pretraining as well as the vast room for continued innovation. Expanding compute power, data, and model architectures promises to further advance protein language modeling and structure prediction into the future.</p> <h2 id="references">References</h2> <ol> <li><a href="https://www.science.org/doi/10.1126/science.ade2574" target="_blank" rel="noopener noreferrer">Evolutionary-scale prediction of atomic-level protein structure with a language model</a></li> <li><a href="https://proceedings.mlr.press/v139/rao21a.html" target="_blank" rel="noopener noreferrer">MSA Transformer</a></li> <li><a href="https://www.nature.com/articles/s41586-021-03819-2/" target="_blank" rel="noopener noreferrer">Highly accurate protein structure prediction with AlphaFold</a></li> <li><a href="https://arxiv.org/abs/1706.03762" target="_blank" rel="noopener noreferrer">Attention is all you need</a></li> <li><a href="https://www.semanticscholar.org/paper/Language-Models-are-Unsupervised-Multitask-Learners-Radford-Wu/9405cc0d6169988371b2755e573cc28650d14dfe" target="_blank" rel="noopener noreferrer">Language Models are Unsupervised Multitask Learners</a></li> <li><a href="https://www.pnas.org/doi/10.1073/pnas.1111471108" target="_blank" rel="noopener noreferrer">Direct-coupling analysis of residue coevolution captures native contacts across many protein families</a></li> </ol> </d-article> <d-appendix> <d-footnote-list></d-footnote-list> <d-citation-list></d-citation-list> </d-appendix> </div> <footer class="fixed-bottom"> <div class="container mt-0"> Â© Copyright 2026 Amitesh Badkul. </div> </footer> <d-bibliography src="/assets/bibliography/"></d-bibliography> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> </body> </html>